packages/carousel/src/index.js

```diff
@@ -0,0 +1,228 @@
+      <Dot
+        key={mapIndex}
+        mx={2}
+        mt={2}
+        bg={mapIndex === currentItemIndex ? 'blue' : 'borderGray'}
+        onClick={this.updateSelectedItem.bind(this, mapIndex)}
+      />
+    ))
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({
+      items: nextProps.children
+    })
+  }
+
+  render() {
+    const { currentItemTranslate, isSwiping } = this.state
+
+    const { shouldShowArrows } = this.props
+
+    const currentItemIndex =
+      this.props.currentItemIndex || this.state.internalCurrentItemIndex
+
+    return (
+      <Box {...this.props}>
+        <OverflowContainer w={1} mb={3}>
+          {shouldShowArrows && (
+            <div>
+              <PositionedArrowLeftController
+                handleClick={this.handleSwipedRight}
+                iconDirection="chevronLeft"
+              />
+              <PositionedArrowRightController
+                handleClick={this.handleSwipedLeft}
+                iconDirection="chevronRight"
+              />
+            </div>
+          )}
+          <Swipeable
+            onSwipingLeft={this.handleSwipingLeft}
+            onSwipingRight={this.handleSwipingRight}
+            onSwipedLeft={this.handleSwipedLeft}
+            onSwipedRight={this.handleSwipedRight}
+            trackMouse
+          >
+            <SwiperWrap
+              translateX={-92 * currentItemIndex + currentItemTranslate + 4}
+              isSwiping={isSwiping}
+            >
+              {this.mapItems()}
+            </SwiperWrap>
+          </Swipeable>
+        </OverflowContainer>
+        <Flex wrap justify="center">
+          {this.mapDots()}
+        </Flex>
+      </Box>
+    )
+  }
+}
+
+const OverflowContainer = styled(Relative)`
+  overflow: hidden;
+`
+
+const Dot = styled(Box)`
+  height: 12px;
+  width: 12px;
+  border-radius: 50%;
+  cursor: pointer;
+`
+
+Dot.displayName = 'Dot'
+
+const PositionedArrowLeftController = styled(ArrowController)`
+  position: absolute;
+  top: 38%;
+  z-index: 1;
+`
+
+const PositionedArrowRightController = styled(PositionedArrowLeftController)`
+  right: 0;
+`
+
+const SwiperWrap = styled.div`
+  transition-property: all;
+  transition-duration: ${props => (props.isSwiping ? '0s' : '0.25s')};
+  transition-timing-function: ease-in-out;
+  transition-delay: initial;
+  left: 0;
+  right: 0;
+  top: 0;
+  bottom: 0;
+  padding: 0;
+  white-space: nowrap;
+  transform: ${props => `translate(${props.translateX}%, 0)`};
+`
+
+const PositionItem = styled(Box)`
+  display: inline-flex;
+  width: 92%;
+  object-fit: contain;
+
+  transition: opacity 0.3s ease 0s;
+  user-select: none;
+  * {
+    user-drag: none;
+  }
+
+  > * {
+    width: 100%;
+    white-space: normal;
+  }
+`
+
+// Carousel.propTypes = {
+//   children: PropTypes.arrayOf(PropTypes.node).isRequired,
+//   currentItemIndex: PropTypes.number,
+//   shouldShowArrows: PropTypes.bool
+// }
+
+Carousel.defaultProps = {
+  children: []
```

Comment:
It's better to make use of `React.Children.toArray` to ensure that `props.children` is an array